src/components/ifxRange/ifxRange.tsx

```diff
@@ -0,0 +1,85 @@
+import { Component, h, Prop, Event, EventEmitter } from '@stencil/core';
+
+@Component({
+  tag: 'ifx-range',
+  styleUrl: 'ifxRange.scss',
+  shadow: true,
+})
+export class IfxRange {
+  @Prop() min: number = 0;
+  @Prop() max: number = 100;
+  @Prop({ mutable: true }) value: number = 0;
+  @Prop() disabled: boolean = false;
+  @Prop() showPercentage: boolean = false; // New property for toggleable percentage display
+  @Prop() leftIcon: string;
+  @Prop() rightIcon: string;
+  @Prop() leftText: string;
+  @Prop() rightText: string;
+  @Event() valueChanged: EventEmitter<number>;
+
+  private inputRef: HTMLInputElement;
+
+  handleInputChange(event: Event) {
+    const target = event.target as HTMLInputElement;
+    this.value = parseInt(target.value);
+    this.valueChanged.emit(this.value);
+    this.updateValuePercent();
+  }
+
+  updateValuePercent() {
+    if (this.inputRef) {
+      const percent = ((this.value - this.min) / (this.max - this.min)) * 100;
```

Comment:
Why not just use the `this.value`? 
`((this.value - this.min) / (this.max - this.min)) * 100` is equivalent to `this.value`.
`const percent = this.value` gives you the same result.

Reply:
we have configurable min and max fields -> that means we need to calculate the percentage this way.
Think about min: 50, max: 90, value: 75